sql/catalyst/src/main/scala/org/apache/spark/sql/errors/QueryErrorsBase.scala

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.spark.sql.errors
+
+import org.apache.spark.sql.catalyst.expressions.Literal
+import org.apache.spark.sql.types.StringType
+import org.apache.spark.unsafe.types.UTF8String
+
+trait QueryErrorsBase {
+  // Converts an error class parameter to its SQL representation
+  def toSQLValue(v: Any): String = v match {
+    case _: UTF8String => Literal.create(v, StringType).sql
+    case _ => Literal(v).sql
```

Comment:
how can we distinguish binary type and array<byte> if the input is `Array[Byte]`?

Reply:
maybe `toSQLValue` should take an optional data type parameter

Reply:
Added an alternative method ...